### ISSUE TITLE
DetailsHeader: empty column header should not be announced by screen reader

### ISSUE THREAD

**Issue by kidokidozh** [microsoft/fluentui#18474]
### Environment Information

- **Package version(s)**: Fluent UI React 8.17.1
- **Browser and OS versions**: Microsoft Edge Version 91.0.864.41 (Official build) (64-bit)

### Describe the issue:

When `DetailsHeader` has `selectAllVisibility={SelectAllVisibility.hidden}`, an empty column header is defined in table. While navigating in table/scan mode using screen reader, focus will go to the empty column and announcing "column header 1 of 10".

![image](https://user-images.githubusercontent.com/58833558/121141132-36b8c000-c86d-11eb-848c-0e850696fe9b.png)

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/kidokidozh/pen/PopagXK

#### Expected behavior:

Empty column header should not be announced by screen reader.

**Comment by gouttierre** [comment]
@kidokidozh - thanks for filing this issue with us.

@smhigley / @ling1726 -  Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by smhigley** [comment]
The current behavior is the correct one. Even without text the column still exists, so the column's header needs to exist whether or not it has content. When there is no next within a cell (header or otherwise), announcing it as blank is the best thing to do.

Tables have a very particular structure, and cells can't be removed without breaking that. Sighted users get to see an empty cell there, so screen reader users should be given that information as well.
